### PR TITLE
Ensure deleting an experiment deletes the associated FF

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

When deleting an experiment, ensure we soft delete the FF as well. (A reverse cascade, hence not covered in the django model cascade option)

## How did you test this code?

Unit tests

### CODE REVIEW

File: posthog/api/feature_flag.py [hunk 1]
```diff
@@ -267,5 +270,10 @@ def my_overrides(self, request: request.Request, **kwargs):
             return Response(serializer.data, status=status.HTTP_201_CREATED)
 
 
+@receiver(pre_delete, sender=Experiment)
```
Comment (macobo) [inline posthog/api/feature_flag.py:273]: Shouldn't this be in the model? :thinking: 

Very hard to find this behavior otherwise.

Comment (neilkakkar) [reply posthog/api/feature_flag.py:273]: makes sense!

**Comment by neilkakkar** [comment]
Ah, just realised that soft-deleting the FF means the unique-key constraint would still persist, which is annoying. Will switch this around to be a hard delete instead, when experiment is deleted.

Edit: nope, all good! These are hard deleted on creation with new same key. :)

**Review by liyiy — APPROVED** [review]

**Comment by neilkakkar** [comment]
okay, this -> https://github.com/PostHog/posthog/pull/7981#issuecomment-1010083793 -> was indeed problematic since the experiment FF creation wasn't using the serializer which has the logic for making this happen.

Instead of trying to wrap this again into `object.create()`, went for using the serializer instead! (and made downstream fixes to the rollout %s)

**Pull request merged**